ssl/ssl_transfer_asn1.cc

```diff
@@ -144,48 +183,100 @@ static const unsigned kS3NotResumableTag =
 // ssl3_state_to_bytes serializes |in| to bytes stored in |cbb|.
 // It returns one on success and zero on failure.
 //
-// An SSL3_STATE is serialized as the following ASN.1 structure:
+// An SSL3_STATE is serialized as the following ASN.1 structure, a complete
+// description can be found in tls_transfer.asn:
+//
+// SSL3StateSerializationVersion ::= INTEGER {
+//                   v1 (1),
+//                   v2 (2) -- Added additional fields to support TLS 1.3
```

Comment:
Are you checking that v1 serialised data can be deserialised by v2 decoder? And error path on the opposite direction?

Reply:
The `kEncodeDecodeKATs` tests have a case for this already (deserializing v1 encoded data successfully). Existing code versions will fail to decode the v2 format per this check https://github.com/aws/aws-lc/blob/main/ssl/ssl_transfer_asn1.cc#L349